MsCorePkg/MacAddressEmulationDxe/MacAddressEmulationDxe.c

```diff
@@ -0,0 +1,239 @@
+/** @file
+
+  DXE Driver for handling MAC Address Emulation.
+
+  Copyright (C) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include "MacAddressEmulationDxe.h"
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/NetLib.h>
+
+#include "Library/MacAddressEmulationPlatformLib.h"
+
+
+BOOLEAN
+SnpSupportsMacEmuCheck (
+  IN CONST EFI_HANDLE SnpHandle,
+  IN CONST EFI_SIMPLE_NETWORK_PROTOCOL *Snp,
+  IN CONST VOID *SnpContext
+  )
+{
+  MAC_EMULATION_SNP_NOTIFY_CONTEXT* Context;
+  BOOLEAN IsMatch = FALSE;
+
+  if (SnpHandle != NULL && Snp != NULL && SnpContext != NULL) {
```

Comment:
I would suggest using a return here.  
Yes have multiple return points from a function can be a bad design but this design leaves a lot of room for mistakes and is much harder to read.   For basic parameter checks I would suggest using a return.